scio-core/src/main/scala/com/spotify/scio/values/PCollectionWrapper.scala

```diff
@@ -42,7 +42,8 @@ private[values] trait PCollectionWrapper[T] extends TransformNameable {
     transform: PTransform[_ >: PCollection[T], PCollection[U]]
   ): SCollection[U] = {
     val t =
-      if (classOf[Combine.Globally[T, U]] isAssignableFrom transform.getClass) {
+      if ((classOf[Combine.Globally[T, U]] isAssignableFrom transform.getClass)
+          && internal.getWindowingStrategy != WindowingStrategy.globalDefault()) {
```

Comment:
What's the reasoning behind modifying this `PTransform` in Scio, instead of letting Beam throw an error upon [expansion of the Combine.Globally transform](https://github.com/apache/beam/blob/c2f0d282337f3ae0196a7717712396a5a41fdde1/sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Combine.java#L1184)? Since we're doing this modification "behind the scenes" users might get confused why their code isn't behaving like they'd expect?

Reply:
Without this, all Scio "global combine" methods will fail when there's windowing and there's no way to fix without dropping to Beam API.

Reply:
cool, I was forgetting that user code doesn't directly interact directly with `pApply`. in that case isn't it redundant to add `.withoutDefaults())` to global combines elsewhere in this PR?

Reply:
Hmm... They're not exactly the same, depending on windowing.

With windowing:
- `withoutDefaults` always needed, regardless whether the `fn` supports default value or not.

Without windowing:
- `aggregate(Aggregator)`, `combine`, `sum` - always need `withoutDefaults`, otherwise the runner might attempt to get default value and fail the assertion below.
`at com.spotify.scio.util.Functions$$anon$2.extractOutput(Functions.scala:171)`
- `fold`, `aggregate(zeroValue)` provides defaults and should not call `withoutDefaults` unless if there's windowing.

So basically it depends on the type of `CombineFn` which we can't determine here.